universal-application-tool-0.0.1/app/services/cloud/StorageClient.java

```diff
@@ -0,0 +1,12 @@
+package services.cloud;
+
+import java.net.URL;
```

Comment:
something to be aware of: `java.net.URL` has some pretty severe gotchas to look out for. In general, never put one in a hash table or use its equality comparison. Either operation triggers a blocking network request to perform DNS resolution. More info here: https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/net/URL.html#equals(java.lang.Object)

Reply:
Ack thanks